scripts/inference.py

```diff
@@ -355,9 +373,47 @@
     # [ATTENTION] for GPTQ on AIU, we must always instantiate an unfused
     # model, the adapter will take care of converting key/values from
     # ckpt into the appropriate form for the model
-    if fused_weights:
+    if fused_weights and is_aiu_backend:
         raise ValueError("GPTQ checkpoints on AIU must always run with --unfuse_weights")
     default_dtype = None  # GPTQ dtype always comes from ckpt, can't be enforced
+elif args.quantization == "int8":
+    def select_int8_module(
+        module_name: str | None = None,
+        smoothquant: bool = True,
+        smoothquant_layers: list[str] | None = None,
+    ):
+        if module_name is None:
+            return "int8_aiu"
+        smoothquant_on_module = (
+            any([m in module_name for m in smoothquant_layers])
+            if smoothquant_layers is not None
+            else True
+        )
+        use_smoothquant = smoothquant and smoothquant_on_module
+        return "int8_smoothquant_aiu" if use_smoothquant else "int8_aiu"
+
+    if args.smoothquant:
+        if any("granite" in p for p in [args.model_path, args.architecture]):
+            smoothquant_layers = ["key", "value", "w1", "wg"]
+        elif any("roberta" in p for p in [args.model_path, args.architecture]):
+            smoothquant_layers = ["query", "key", "value", "w1"]
+        else:
+            raise NotImplementedError("INT8 architecture not supported.")
+    else:
+        smoothquant_layers = []
+
+    linear_config = {
+        "linear_type": partial(
+            select_int8_module,
+            smoothquant = args.smoothquant,
+            smoothquant_layers = smoothquant_layers,
+        ),
+        "weight_per_channel": args.weight_per_channel,
+        "activ_quant_type": args.activ_quant_type,
+    }
+    if fused_weights and is_aiu_backend:
+        raise ValueError("INT8 checkpoints on AIU must always run with --unfuse_weights")
+    default_dtype = None  # INT8 dtype always comes from ckpt
```

Comment:
just assert this is None and raise error for user instead of silently changing the user parameter